From 5cf70302edca2b7c61c6c5a6ebdd7c89d37e3c6c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 31 Oct 2025 15:21:39 +0100
Subject: [PATCH] Fix some colors in the built-in theme

This fixes the colors when a light GTK theme is used with the built-in
theme. Without this, dark icons/text are displayed over dark background at
the following places:
- Session and polkit dialogs text and icon.
- Main menu footer and overlay menu icons.
- Raven notifications view placeholder text and icon.
---
 src/theme/common/_dialogs.scss | 1 +
 src/theme/common/_menu.scss    | 2 ++
 src/theme/common/_raven.scss   | 4 ++++
 3 files changed, 7 insertions(+)

diff --git a/src/theme/common/_dialogs.scss b/src/theme/common/_dialogs.scss
index 7215e004a..c72af0af4 100644
--- a/src/theme/common/_dialogs.scss
+++ b/src/theme/common/_dialogs.scss
@@ -1,5 +1,6 @@
 // Base class for dialogs
 @mixin budgie_dialog() {
+    color: $fg_color;
     background-color: transparent;
     font-weight: normal;
 
diff --git a/src/theme/common/_menu.scss b/src/theme/common/_menu.scss
index 0411649e0..c7172436e 100644
--- a/src/theme/common/_menu.scss
+++ b/src/theme/common/_menu.scss
@@ -87,6 +87,7 @@
         .image-button {
             @extend %reset_style;
 
+            color: $fg_color;
             transition-property: background-color;
             transition: 170ms ease-out;
             padding: 4px;
@@ -122,6 +123,7 @@
         button {
             @extend %reset_style;
 
+            color: $fg_color;
             transition-property: background-color, color;
             transition: 170ms ease-out;
             padding: 4px 8px 4px 2px;
diff --git a/src/theme/common/_raven.scss b/src/theme/common/_raven.scss
index 58653d5aa..f0b629b91 100644
--- a/src/theme/common/_raven.scss
+++ b/src/theme/common/_raven.scss
@@ -31,6 +31,10 @@
     .raven-notifications-view > .raven-background > .frame > list {
         @extend %reset_style;
 
+        .dim-label {
+            color: $fg_color;
+        }
+
         .raven-notifications-group {
             .raven-notifications-group-header {
                 padding: 0 4px;
